src/app/(dashboard)/layout.tsx

```diff
@@ -108,7 +108,7 @@ export default async function DashboardLayout({
     return (
       <HydrationBoundary state={dehydratedState}>
         <Fragment>
-          <DashboardHeader />
+          <ConditionalHeader />
```

Comment:
I'd say move the logic for the conditional header into this file, just because it is a single line, and semantically, it makes more sense to render something conditionally from this component as opposed to always rendering it, but with the conditional part inside ConditionalHeader.